Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/manager_app/view/plan_create_task_list/PlanCreateTaskListRecyclerViewAdapter.java

```diff
@@ -0,0 +1,167 @@
+package pg.autyzm.friendly_plans.manager_app.view.plan_create_task_list;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import database.entities.PlanTaskTemplate;
+import database.entities.TaskTemplate;
+import pg.autyzm.friendly_plans.R;
+import pg.autyzm.friendly_plans.asset.AssetsHelper;
+import pg.autyzm.friendly_plans.item_touch_helper.ItemTouchHelperAdapter;
+
+public class PlanCreateTaskListRecyclerViewAdapter extends
+        RecyclerView.Adapter<PlanCreateTaskListRecyclerViewAdapter.TaskListViewHolder>  implements ItemTouchHelperAdapter {
+
+    private static final int ICON_PLACEHOLDER_PICTURE_ID = R.drawable.ic_placeholder;
+    private static final int ICON_PLACEHOLDER_SOUND_ID = R.drawable.ic_playing_sound;
+    private static final int ICON_PLACEHOLDER_TIME_ID = R.drawable.ic_placeholder_time;
+    private static List<PlanTaskTemplate> planTaskItemList = new ArrayList<>();
+    private TaskItemClickListener taskItemClickListener;
+    private AssetsHelper assetsHelper;
+
+    PlanCreateTaskListRecyclerViewAdapter(TaskItemClickListener taskItemClickListener) {
+        this.taskItemClickListener = taskItemClickListener;
+    }
+
+    @Override
+    public boolean onItemMove(int fromPosition, int toPosition) {
+        if (fromPosition < toPosition) {
```

Comment:
Same code is in steps list. It would be good to extract it so some common component like. ItemOrderUpdater with method swapOrder(list, from, to)